includes/class-nonfig-wp-api-activator.php

```diff
@@ -31,36 +31,37 @@ class Nonfig_Wp_Api_Activator {
 	 */
 	public static function activate() {
 
-	}
 
 
-	public static function nonfig_cache_db_install() {
-		global $wpdb;
-		$table_name      = 'nonfig_' . $wpdb->prefix . 'cache';
-		$charset_collate = $wpdb->get_charset_collate();
-		$sql             = "CREATE TABLE $table_name (
+        global $wpdb;
+        $table_name = 'nonfig_' . $wpdb->prefix . "cache";
+        $charset_collate = $wpdb->get_charset_collate();
+        $sql = "CREATE TABLE $table_name (
               id mediumint(9) NOT NULL AUTO_INCREMENT,
               time datetime DEFAULT '0000-00-00 00:00:00' NOT NULL,
-              cache_dur datetime DEFAULT '0000-00-00 00:00:00' NOT NULL,
               nonfig_value varchar(128) NOT NULL default '',
+              nonfig_result text NOT NULL default '',
```

Comment:
```
id: UUID
config_id: UUID
config_name: string // config.name
config_path: string // config.fullyQualifiedPath
createdAt: Datetime
```

this would be enough. Whenever we fetch configurations, just store them over here. Everytime we need to fetch the same config, just check for `createdAt` and see if it is within cache duration.

if yes: pick from db and return
if no: fetch from server(), upsert in db and return